lib/authentication/secure_storage/json_credential_manager.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const path = require('path');
+const Logger = require('../../logger');
+const fs = require('fs');
+const os = require('os');
+const Util = require('../../util');
+
+function JsonCredentialManager() {
+
+  this.userinfo = os.userInfo();
+
+  this.getTokenDir = function () {
+    let tokenDir = process.env.SF_TEMPORARY_CREDENTIAL_CACHE_DIR;
+
+    if (!Util.exists(tokenDir)) {
+      tokenDir = os.homedir();
+    } else {
+      Logger.getInstance().info(`The credential cache directory is configured by the user. The token will be saved at ${tokenDir}`);
+    }
+
+    if (!Util.exists(tokenDir)) {
+      Logger.getInstance().error(`Temporary credential cache directory is invalid, and the driver is unable to use the default location(home). 
+      Please assign the enviroment variable value SF_TEMPORARY_CREDENTIAL_CACHE_DIR to enable the default credential manager.`);
+      return null; 
+    }
+    const tokenCacheFile = path.join(tokenDir, 'temporary_credential.json');
+
+    if (fs.existsSync(tokenCacheFile)) {
```

Comment:
should we use sync methods? can we switch to async ones?

Reply:
If we should use the async functions, I think the sync functions are okay for the condition check. Is there any other reason why we need to use async functions?

Reply:
in the nodejs driver we would like to use as much as possible the nonblocking code - let's avoid the sync methods

But if we really really need to have it then let it be cc @sfc-gh-pmotacki

Reply:
I agree. When we create new piece of code the general rule is to write unblocking code. Please replace to async methods where it is possible.

Reply:
I replaced all the sync functions to the asnyc functions. Please have a look.